<!DOCTYPE html>
<html>
  <head>
    <title>测试报告</title>
    <meta charset="UTF-8">
    <style>body {    
    font-family: consolas, Verdana, sans-serif;
    font-size: .92em;
    color: #696e71;
}

.main_section {   
    width: 90%;
    margin: 0 auto;
}

#float_menu{    
    position:fixed;
    top:0;
    right:0;
    text-align: center;
}

#float_menu .menu-item {       
    cursor: pointer;
    padding: .5em;
    margin: .5em 0;
    color: #c08580;
    background-color: #f8f0ef;
    font-size: 1.2em;
}

table.result_stats{
    border-collapse: collapse;
    border: 1px solid #f0e0e5;
    width: 35em;
    text-align: center;
    font-size: 1.0em;
}
    
table.result_stats td{
    border: 1px solid #f0e0e5;
    padding: .3em;
}

.h3_button {
    margin: 1.5em;
    cursor: pointer;
    color: #03a9f4;
}


.info
{
    white-space:pre-wrap;
    padding: .6em;
}


.error-info
{
    color: #a64747
}

.suite_dir {
    margin: 1em .2em;
    padding: .3em;
    /* background-color: #dfeff6; */
    border: 1px solid #bcd8e4;
}
.suite_file {
    margin: 1em .2em;
    padding: .3em;
    border: 1px solid #bcd8e4;
}


.case {
    margin: 1em .2em;
    /* padding: .3em; */
    border: 1px solid #e7d4d4;
}

.case_class_path{
    margin: 0em 1em; 
}


.folder_header { 
    padding: .2em .7em;
    background-color: #fffaf9;
    cursor: pointer;
}


.setup{
    margin: .2em;
    padding: .3em;
    /* border: 1px solid #e7d4d4; */
}
.teardown{
    margin: .2em;
    padding: .3em;
    /* border: 1px solid #e7d4d4; */
}
.test_steps{
    margin: .2em;
    padding: .3em;
    /* border: 1px solid #e7d4d4; */
}


.label {
    display: inline-block;
    padding: .1em .5em;
    font-size: .88em;
    letter-spacing: 1px;
    white-space: nowrap;
    color: #0d6ebc;
    border-radius: .2em;
    min-width: 5em;    
    margin-right: 2em;
    font-family: consolas;
}

/* .suite_setup .label{
    color: #219e26 ;
}
.suite_teardown .label{
    color: #219e26;
} */


/* .case.pass   .casename{
    color: #329132 ;
} */
.case.pass   .caselabel{
    color: white;
    background-color: #3b9e3f;
}
/* .case.fail   .casename{
    color: #a64747;
} */
.case.fail   .caselabel{
    color: white;
    background-color: #a64747;
}
/* .case.abort   .casename{
    color: #953ab7;
} */
.case.abort   .caselabel{
    color: white;
    background-color: #9c27b0;
}



.case_step  {
    margin: .8em;
}
.checkpoint_pass {
    margin: .8em;
}
.checkpoint_fail {
    margin: .8em;
}

.case_step  .tag{
    color: #2196f3;;
    margin: .3em 1em .3em 0;
    padding: .1em .3em;
    font-size: .92em;
}

.checkpoint_pass .tag{
    color: #009806;
    margin:.3em 1em .3em .5em;
    padding: .1em .3em;
    font-size: .92em;
}
.checkpoint_fail .tag{
    color: #9c2020;
    margin:.3em 1em .3em .5em;
    padding: .1em .3em;
    font-size: .92em;
}</style>
    <script type="text/javascript">var FOLDER_ALL_CASES = false //是否为精简模式的标记
var ERROR_INFOS = [];  // 错误信息列表
var current_error_idx = -1;

// 页面加载后执行的函数
window.addEventListener("load", function(){
    // 所有 .folder_header 添加点击事件处理
    let folderHeaderEles = document.querySelectorAll(".folder_header");
    folderHeaderEles.forEach(function(ele) {
        ele.addEventListener("click", function(event) {
        let fb = event.target.closest('.folder_header').nextElementSibling;
        fb.style.display = fb.style.display === 'none' ? 'block' : 'none'
        });
    });

    // 找到所有的错误信息对象
    ERROR_INFOS = document.querySelectorAll(".error-info");
});



function toggle_folder_all_cases(){
    let eles = document.querySelectorAll(".folder_body");
    
    FOLDER_ALL_CASES = !FOLDER_ALL_CASES;
    document.getElementById('display_mode').innerHTML = FOLDER_ALL_CASES? "详细" : "精简"

    for (const ele of eles){
        ele.style.display =  FOLDER_ALL_CASES? "none": "block"
    }
    
}



function previous_error(){
    // 查找错误必须是详细模式
    if (FOLDER_ALL_CASES)
        toggle_folder_all_cases()

    current_error_idx -= 1; 
    if (current_error_idx<0)
        current_error_idx = 0

    
    let error = ERROR_INFOS[current_error_idx];

    error.scrollIntoView({behavior: "smooth", block: "center", inline: "start"});

    
}


function next_error(){
    
    // 查找错误必须是详细模式
    if (FOLDER_ALL_CASES)
        toggle_folder_all_cases()

    current_error_idx += 1;
    if (current_error_idx > ERROR_INFOS.length-1)
        current_error_idx = ERROR_INFOS.length-1

    let error = ERROR_INFOS[current_error_idx];

    error.scrollIntoView({behavior: "smooth", block: "center", inline: "start"});
    
}</script>
  </head>
  <body>
    <div class="main_section">
      <h1 style="font-family: auto">测试报告 - hytest v0.5.9</h1>
      <h3>统计结果</h3>
      <table class="result_stats">
        <tbody>
          <tr>
            <td>开始时间</td>
            <td>20210315 10:49:51</td>
          </tr>
          <tr>
            <td>结束时间</td>
            <td>20210315 10:50:23</td>
          </tr>
          <tr>
            <td>耗时</td>
            <td>32.456 秒</td>
          </tr>
          <tr>
            <td>用例数量</td>
            <td>4</td>
          </tr>
          <tr>
            <td>通过</td>
            <td>2</td>
          </tr>
          <tr>
            <td>失败</td>
            <td style="">0</td>
          </tr>
          <tr>
            <td>异常</td>
            <td style="color:red">2</td>
          </tr>
          <tr>
            <td>套件初始化失败</td>
            <td style="">0</td>
          </tr>
          <tr>
            <td>套件清除失败</td>
            <td style="">0</td>
          </tr>
          <tr>
            <td>用例初始化失败</td>
            <td style="">0</td>
          </tr>
          <tr>
            <td>用例清除失败</td>
            <td style="">0</td>
          </tr>
        </tbody>
      </table>
      <div style="margin-top:2em">
        <h3 style="display:inline">执行日志</h3>
      </div>
      <div class="exec_log">
        <div class="suite_dir" id="suite_dir cases\">
          <div>
            <span class="label">进入目录</span>
            <span>cases\</span>
          </div>
          <div class="suite_setup setup" id="suite_setup setup cases\">
            <div class="folder_header">
              <span class="label">套件初始化</span>
              <span>cases\</span>
            </div>
            <div class="folder_body">
              <div class="info">suite_setup</div>
              <div class="info">打开浏览器</div>
            </div>
          </div>
        </div>
        <div class="suite_file" id="suite_file cases\管理员操作.py">
          <div>
            <span class="label">进入文件</span>
            <span>cases\管理员操作.py</span>
          </div>
          <div class="suite_setup setup" id="suite_setup setup cases\管理员操作.py">
            <div class="folder_header">
              <span class="label">套件初始化</span>
              <span>cases\管理员操作.py</span>
            </div>
            <div class="folder_body">
              <div class="info">suite_setup</div>
              <div class="info">打开浏览器</div>
            </div>
          </div>
          <div class="case pass" id="case_00000001">
            <div class="folder_header">
              <span class="label caselabel">用例 PASS</span>
              <span class="casename">检查操作菜单 UI_0101</span>
            </div>
            <div class="folder_body">
              <span class="case_class_path">cases\管理员操作.py::UI_0101</span>
              <div class="test_steps" id="test_steps 检查操作菜单 UI_0101">
                <span class="label">测试步骤</span>
                <div class="case_step">
                  <span class="tag">第 1 步</span>
                  <span>获取左侧菜单信息</span>
                </div>
                <div class="info">['客户', '药品', '订单', '其它菜单', '']</div>
                <div class="case_step">
                  <span class="tag">第 2 步</span>
                  <span>检查菜单栏</span>
                </div>
                <div class="checkpoint_pass">
                  <span class="tag">检查点 PASS</span>
                  <span>左侧菜单检查</span>
                </div>
              </div>
            </div>
          </div>
          <div class="case pass" id="case_00000002">
            <div class="folder_header">
              <span class="label caselabel">用例 PASS</span>
              <span class="casename">添加客户 UI_0102</span>
            </div>
            <div class="folder_body">
              <span class="case_class_path">cases\管理员操作.py::UI_0102</span>
              <div class="test_steps" id="test_steps 添加客户 UI_0102">
                <span class="label">测试步骤</span>
                <div class="case_step">
                  <span class="tag">第 1 步</span>
                  <span>点击左侧客户菜单</span>
                </div>
                <div class="case_step">
                  <span class="tag">第 2 步</span>
                  <span>添加客户</span>
                </div>
                <div class="case_step">
                  <span class="tag">第 3 步</span>
                  <span>检查添加信息</span>
                </div>
                <div class="info">['客户名：', '南京中医院', '联系电话：', '2551867858', '地址：', '江苏省-南京市-秦淮区-汉中路-16栋504']</div>
                <div class="checkpoint_pass">
                  <span class="tag">检查点 PASS</span>
                  <span>客户信息和添加内容一致 </span>
                </div>
              </div>
            </div>
          </div>
          <div class="suite_teardown teardown" id="suite_teardown teardown cases\管理员操作.py">
            <div class="folder_header">
              <span class="label">套件清除</span>
              <span>cases\管理员操作.py</span>
            </div>
            <div class="folder_body">
              <div class="info">suite_teardown</div>
            </div>
          </div>
        </div>
        <div class="suite_file" id="suite_file cases\管理员操作2.py">
          <div>
            <span class="label">进入文件</span>
            <span>cases\管理员操作2.py</span>
          </div>
          <div class="case abort" id="case_00000003">
            <div class="folder_header">
              <span class="label caselabel">用例 ABORT</span>
              <span class="casename">检查操作菜单 UI_0101</span>
            </div>
            <div class="folder_body">
              <span class="case_class_path">cases\管理员操作2.py::UI_0101</span>
              <div class="test_steps" id="test_steps 检查操作菜单 UI_0101">
                <span class="label">测试步骤</span>
                <div class="case_step">
                  <span class="tag">第 1 步</span>
                  <span>获取左侧菜单信息</span>
                </div>
                <div class="info error-info">HTTPConnectionPool(host='127.0.0.1', port=52625): Max retries exceeded with url: /session/03ff248b87bcb3eab2559f0fc66c250c/elements (Caused by NewConnectionError('&lt;urllib3.connection.HTTPConnection object at 0x0000023716106E48&gt;: Failed to establish a new connection: [WinError 10061] 由于目标计算机积极拒绝，无法连接。',)) 
Traceback:
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\util\connection.py&quot;, line 79, in create_connection
    raise err
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\util\connection.py&quot;, line 69, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [WinError 10061] 由于目标计算机积极拒绝，无法连接。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 600, in urlopen
    chunked=chunked)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 354, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File &quot;d:\tools\python3.6.5\lib\http\client.py&quot;, line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File &quot;d:\tools\python3.6.5\lib\http\client.py&quot;, line 1285, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File &quot;d:\tools\python3.6.5\lib\http\client.py&quot;, line 1234, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File &quot;d:\tools\python3.6.5\lib\http\client.py&quot;, line 1026, in _send_output
    self.send(msg)
  File &quot;d:\tools\python3.6.5\lib\http\client.py&quot;, line 964, in send
    self.connect()
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connection.py&quot;, line 196, in connect
    conn = self._new_conn()
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connection.py&quot;, line 180, in _new_conn
    self, &quot;Failed to establish a new connection: %s&quot; % e)
urllib3.exceptions.NewConnectionError: &lt;urllib3.connection.HTTPConnection object at 0x0000023716106E48&gt;: Failed to establish a new connection: [WinError 10061] 由于目标计算机积极拒绝，无法连接。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File &quot;d:\tools\python3.6.5\lib\site-packages\hytest\utils\runner.py&quot;, line 161, in _exec_cases
    try:case.teststeps();signal.case_pass(cls.caseId,case.name)
  File &quot;cases\管理员操作2.py&quot;, line 13, in teststeps
    eles = wd.find_elements_by_css_selector('.sidebar-menu li span')
  File &quot;d:\tools\python3.6.5\lib\site-packages\selenium\webdriver\remote\webdriver.py&quot;, line 614, in find_elements_by_css_selector
    return self.find_elements(by=By.CSS_SELECTOR, value=css_selector)
  File &quot;d:\tools\python3.6.5\lib\site-packages\selenium\webdriver\remote\webdriver.py&quot;, line 1007, in find_elements
    'value': value})['value'] or []
  File &quot;d:\tools\python3.6.5\lib\site-packages\selenium\webdriver\remote\webdriver.py&quot;, line 319, in execute
    response = self.command_executor.execute(driver_command, params)
  File &quot;d:\tools\python3.6.5\lib\site-packages\selenium\webdriver\remote\remote_connection.py&quot;, line 374, in execute
    return self._request(command_info[0], url, body=data)
  File &quot;d:\tools\python3.6.5\lib\site-packages\selenium\webdriver\remote\remote_connection.py&quot;, line 397, in _request
    resp = self._conn.request(method, url, body=body, headers=headers)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\request.py&quot;, line 72, in request
    **urlopen_kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\request.py&quot;, line 150, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\poolmanager.py&quot;, line 322, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 667, in urlopen
    **response_kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 667, in urlopen
    **response_kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 667, in urlopen
    **response_kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\util\retry.py&quot;, line 398, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='127.0.0.1', port=52625): Max retries exceeded with url: /session/03ff248b87bcb3eab2559f0fc66c250c/elements (Caused by NewConnectionError('&lt;urllib3.connection.HTTPConnection object at 0x0000023716106E48&gt;: Failed to establish a new connection: [WinError 10061] 由于目标计算机积极拒绝，无法连接。',))
</div>
              </div>
            </div>
          </div>
          <div class="case abort" id="case_00000004">
            <div class="folder_header">
              <span class="label caselabel">用例 ABORT</span>
              <span class="casename">添加客户 UI_0102</span>
            </div>
            <div class="folder_body">
              <span class="case_class_path">cases\管理员操作2.py::UI_0102</span>
              <div class="test_steps" id="test_steps 添加客户 UI_0102">
                <span class="label">测试步骤</span>
                <div class="case_step">
                  <span class="tag">第 1 步</span>
                  <span>点击左侧客户菜单</span>
                </div>
                <div class="info error-info">HTTPConnectionPool(host='127.0.0.1', port=52625): Max retries exceeded with url: /session/03ff248b87bcb3eab2559f0fc66c250c/element (Caused by NewConnectionError('&lt;urllib3.connection.HTTPConnection object at 0x000002371619C630&gt;: Failed to establish a new connection: [WinError 10061] 由于目标计算机积极拒绝，无法连接。',)) 
Traceback:
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\util\connection.py&quot;, line 79, in create_connection
    raise err
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\util\connection.py&quot;, line 69, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [WinError 10061] 由于目标计算机积极拒绝，无法连接。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 600, in urlopen
    chunked=chunked)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 354, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File &quot;d:\tools\python3.6.5\lib\http\client.py&quot;, line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File &quot;d:\tools\python3.6.5\lib\http\client.py&quot;, line 1285, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File &quot;d:\tools\python3.6.5\lib\http\client.py&quot;, line 1234, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File &quot;d:\tools\python3.6.5\lib\http\client.py&quot;, line 1026, in _send_output
    self.send(msg)
  File &quot;d:\tools\python3.6.5\lib\http\client.py&quot;, line 964, in send
    self.connect()
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connection.py&quot;, line 196, in connect
    conn = self._new_conn()
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connection.py&quot;, line 180, in _new_conn
    self, &quot;Failed to establish a new connection: %s&quot; % e)
urllib3.exceptions.NewConnectionError: &lt;urllib3.connection.HTTPConnection object at 0x000002371619C630&gt;: Failed to establish a new connection: [WinError 10061] 由于目标计算机积极拒绝，无法连接。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File &quot;d:\tools\python3.6.5\lib\site-packages\hytest\utils\runner.py&quot;, line 161, in _exec_cases
    try:case.teststeps();signal.case_pass(cls.caseId,case.name)
  File &quot;cases\管理员操作2.py&quot;, line 32, in teststeps
    sidebarMenu = wd.find_element_by_class_name('sidebar-menu')
  File &quot;d:\tools\python3.6.5\lib\site-packages\selenium\webdriver\remote\webdriver.py&quot;, line 564, in find_element_by_class_name
    return self.find_element(by=By.CLASS_NAME, value=name)
  File &quot;d:\tools\python3.6.5\lib\site-packages\selenium\webdriver\remote\webdriver.py&quot;, line 978, in find_element
    'value': value})['value']
  File &quot;d:\tools\python3.6.5\lib\site-packages\selenium\webdriver\remote\webdriver.py&quot;, line 319, in execute
    response = self.command_executor.execute(driver_command, params)
  File &quot;d:\tools\python3.6.5\lib\site-packages\selenium\webdriver\remote\remote_connection.py&quot;, line 374, in execute
    return self._request(command_info[0], url, body=data)
  File &quot;d:\tools\python3.6.5\lib\site-packages\selenium\webdriver\remote\remote_connection.py&quot;, line 397, in _request
    resp = self._conn.request(method, url, body=body, headers=headers)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\request.py&quot;, line 72, in request
    **urlopen_kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\request.py&quot;, line 150, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\poolmanager.py&quot;, line 322, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 667, in urlopen
    **response_kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 667, in urlopen
    **response_kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 667, in urlopen
    **response_kw)
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\connectionpool.py&quot;, line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File &quot;d:\tools\python3.6.5\lib\site-packages\urllib3\util\retry.py&quot;, line 398, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='127.0.0.1', port=52625): Max retries exceeded with url: /session/03ff248b87bcb3eab2559f0fc66c250c/element (Caused by NewConnectionError('&lt;urllib3.connection.HTTPConnection object at 0x000002371619C630&gt;: Failed to establish a new connection: [WinError 10061] 由于目标计算机积极拒绝，无法连接。',))
</div>
              </div>
            </div>
          </div>
          <div class="suite_teardown teardown" id="suite_teardown teardown cases\">
            <div class="folder_header">
              <span class="label">套件清除</span>
              <span>cases\</span>
            </div>
            <div class="folder_body">
              <div class="info">suite_teardown</div>
            </div>
          </div>
        </div>
      </div>
      <div id="float_menu">
        <div class="menu-item" onclick="document.querySelector(&quot;body&quot;).scrollIntoView()">页首</div>
        <div class="menu-item" onclick="window.open(&quot;http://www.python3.vip/tut/auto/hytest/01&quot;, &quot;_blank&quot;); ">教程</div>
        <div class="menu-item" id="display_mode" onclick="toggle_folder_all_cases()">精简</div>
        <div class="error_jumper" display="block">
          <div class="menu-item" onclick="previous_error()" title="上一个错误">∧</div>
          <div class="menu-item" onclick="next_error()" title="下一个错误">∨</div>
        </div>
      </div>
    </div>
  </body>
</html>